sound/soc/sof/intel/apl.c

```diff
@@ -131,5 +137,8 @@ const struct sof_intel_dsp_desc apl_chip_info = {
 	.rom_init_timeout	= 150,
 	.ssp_count = APL_SSP_COUNT,
 	.ssp_base_offset = APL_SSP_BASE_OFFSET,
+	.num_hda_drv	= SOF_SKL_NUM_HDA_DAIS,
+	.num_dmic_drv	= SOF_SKL_NUM_DMIC_DAIS,
+	.num_ssp_drv	= SOF_SKL_NUM_SSP_DAIS,
```

Comment:
I think here we can do better by definining a generic dai_drv something like this.

```
struct sof_dai_drv {
      int type;
      int num;
};

then insided sof_intel_dsp_desc we can have an array of sof_dai_drv. Because this is Intel specific I will leave for you to decide.